test/e2e/run-tests.sh

```diff
@@ -35,13 +30,29 @@ if [ ! -d "$service_test_dir" ]; then
     exit 0
 fi
 
+# check to see if the tests use pytest
+[[ -f "$service_test_dir/__init__.py" ]] && enable_python_tests="true" || enable_python_tests="false"
+
 # find all files except under helper directory
 service_test_files=$( find "$service_test_dir" -name helper -prune -false -o -type f ! -name '.*' | sort )
 
-for service_test_file in $service_test_files; do
-    test_name=$( filenoext "$service_test_file" )
-    test_start_time=$( date +%s )
-    bash $service_test_file
-    test_end_time=$( date +%s )
-    echo "$test_name took $( expr $test_end_time - $test_start_time ) second(s)"
-done
+if [[ "$enable_python_tests" == "false" ]]; then
+  for service_test_file in $service_test_files; do
+      test_name=$( filenoext "$service_test_file" )
+      test_start_time=$( date +%s )
+      bash $service_test_file
+      test_end_time=$( date +%s )
+      echo "$test_name took $( expr $test_end_time - $test_start_time ) second(s)"
+  done
+else
+  python bootstrap.py "${SERVICE}"
+
+  set +e
+
+  pushd "$service_test_dir" > /dev/null
+    pytest --log-cli-level INFO .
+  popd > /dev/null
+  python cleanup.py "${SERVICE}"
```

Comment:
Is this guaranteed to be executed if `pytest .` fails?

Reply:
I `set +e` just above this in an attempt to ensure they do. I'm not certain that's foolproof, although I couldn't imagine `pytest` could force an exit.

Reply:
Maybe we refactor the framework a bit (in a future PR) to use pytest.fixture and add finalizers to a module-scoped fixture?

https://docs.pytest.org/en/latest/fixture.html